From 15dc46d1a47708d0ded536aed6dbb350fcde6918 Mon Sep 17 00:00:00 2001
From: Thomas Piccirello <8296030+Piccirello@users.noreply.github.com>
Date: Mon, 16 Jun 2025 11:53:09 -0700
Subject: [PATCH 06/12] WebUI: Increase number of buffered virtual rows

Negligible performance hit for increased UX. Currently when scrolling I frequently see blank rows.

PR #22853.

(cherry picked from commit 380d9af34c02a1f5e5cab83a50f59fa65634cd42)
---
 src/webui/www/private/scripts/dynamicTable.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/webui/www/private/scripts/dynamicTable.js b/src/webui/www/private/scripts/dynamicTable.js
index 2b19f781e..e8afd658b 100644
--- a/src/webui/www/private/scripts/dynamicTable.js
+++ b/src/webui/www/private/scripts/dynamicTable.js
@@ -913,8 +913,8 @@ window.qBittorrent.DynamicTable ??= (() => {
             if (this.dynamicTableDiv.offsetHeight === 0)
                 return;
             this.renderedHeight = this.dynamicTableDiv.offsetHeight;
-            // show extra 6 rows at top/bottom to reduce flickering
-            const extraRowCount = 6;
+            // show extra rows at top/bottom to reduce flickering
+            const extraRowCount = 20;
             // how many rows can be shown in the visible area
             const visibleRowCount = Math.ceil(this.renderedHeight / this.rowHeight) + (extraRowCount * 2);
             // start position of visible rows, offsetted by scrollTop
-- 
2.47.3

